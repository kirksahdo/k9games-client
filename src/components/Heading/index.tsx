import clsx from 'clsx'
import React from 'react'

export interface HeadingProps {
  children: React.ReactNode
  color?: 'white' | 'black'
  line?: 'left' | 'bottom'
  lineColor?: 'primary' | 'secondary'
}

const Heading = ({
  children,
  color = 'black',
  line,
  lineColor = 'primary'
}: HeadingProps) => {
  return (
    <div
      className={clsx(
        'flex flex-col gap-0.75',
        line === 'left' &&
          `border-l-[0.4375rem] border-solid pl-3 border-${lineColor}`
      )}
    >
      <h2
        className={`text-${color} text-xl md:text-2xl font-bold inline-block`}
      >
        {children}
      </h2>
      {line === 'bottom' && (
        <span className={`h-[5px] w-1/3 bg-${lineColor}`}></span>
      )}
    </div>
  )
}

export default Heading
