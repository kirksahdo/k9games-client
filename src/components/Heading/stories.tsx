import { Meta, StoryObj } from '@storybook/react'
import Heading from '.'

export default {
  title: 'Heading',
  component: Heading,
  parameters: {
    layout: 'centered'
  }
} as Meta

type Story = StoryObj<typeof Heading>

export const Default: Story = {
  args: {
    children: 'Heading',
    color: 'white',
    line: undefined,
    lineColor: 'primary'
  }
}
