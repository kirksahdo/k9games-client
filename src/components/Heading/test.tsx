import { render, screen } from '@testing-library/react'
import React from 'react'

import Heading from '.'

describe('<Heading />', () => {
  it('should render a black heading by default', () => {
    const headingText = 'text'
    render(<Heading>{headingText}</Heading>)

    expect(screen.getByRole('heading', { name: headingText })).toHaveClass(
      'text-black'
    )
  })

  it('should render a white heading when color is passed', () => {
    const headingText = 'text'
    render(<Heading color="white">{headingText}</Heading>)

    expect(screen.getByRole('heading', { name: headingText })).toHaveClass(
      'text-white'
    )
  })

  it('should render a heading with a line to the left', () => {
    const headingText = 'text'
    render(<Heading line="left">{headingText}</Heading>)
    const wrapper = screen.getByRole('heading', { name: /text/i }).parentElement

    expect(wrapper).toHaveClass('border-l-[0.4375rem] border-solid pl-3')
    expect(wrapper).toHaveClass('border-primary')
  })

  it('should render a heading with a line at the bottom', () => {
    const headingText = 'text'
    render(<Heading line="bottom">{headingText}</Heading>)
    const wrapper = screen.getByRole('heading', { name: /text/i }).parentElement

    expect(wrapper?.querySelector('span')).toBeInTheDocument()
    expect(wrapper?.querySelector('span')).toHaveClass(
      'h-[5px] w-1/3 bg-primary'
    )
  })

  it('should render a heading with a different line color', () => {
    const headingText = 'text'
    const { rerender } = render(
      <Heading line="left" lineColor="secondary">
        {headingText}
      </Heading>
    )
    const wrapper = screen.getByRole('heading', { name: /text/i }).parentElement
    expect(wrapper).toHaveClass('border-secondary')

    rerender(
      <Heading line="bottom" lineColor="secondary">
        {headingText}
      </Heading>
    )
    expect(wrapper?.querySelector('span')).toHaveClass('bg-secondary')
  })

  it('should render a small heading', () => {
    const headingText = 'text'
    render(<Heading size="small">{headingText}</Heading>)

    expect(screen.getByRole('heading', { name: headingText })).toHaveClass(
      'text-[16px]'
    )
  })
})
