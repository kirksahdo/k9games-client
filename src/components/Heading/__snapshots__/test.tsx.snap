// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading 1`] = `
<div>
  <h1>
    Heading
  </h1>
</div>
`;

exports[`<Heading /> should render the heading text 1`] = `
<h2
  class="text-black text-[1.75rem] font-bold"
>
   
  text
</h2>
`;

exports[`<Heading /> should render the small heading text 1`] = `
<h2
  class="text-black text-[1.375rem] font-bold"
>
   
  text
</h2>
`;
